### PR TITLE
[glsl-new] Update last exprs to ExpressionRule

### PR DESCRIPTION
**Pull request by pjoe**
Refactor out binary_expr to keep parser.rs 'small' and readable

### CODE REVIEW

**Comment by pjoe** [comment]
Hmmm monocodus playing tricks again. I wonder if it is related to those 2 warnings the CI Test on 1.41 also reports 🤔

Anyway I have no clippy issues locally:
```
code@code-server:~/git/naga$ git branch
* glsl-new-expression-rule
  master
  wasm-build
code@code-server:~/git/naga$ cargo clippy --all-features -- -D warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
```

File: src/front/glsl_new/expressions.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use super::ast::*;
+use crate::{BinaryOperator, Expression};
+
+pub fn binary_expr(
+    program: &mut Program,
```
Comment (kvark) [inline src/front/glsl_new/expressions.rs:5]: would it makes sense to have as a mutable method on the `Program`?

Comment (pjoe) [reply src/front/glsl_new/expressions.rs:5]: Yeah, that could make sense. I like the idea of having functionality split into different files like: expressions, types, variables, but right now that may be a bit overkill for a single function.

Comment (pjoe) [reply src/front/glsl_new/expressions.rs]: Tried it out - much simpler :D

File: src/front/glsl_new/ast.rs [hunk 2]
```diff
@@ -39,6 +39,19 @@ impl Program {
             },
         }
     }
+
+    pub fn binary_expr(
+        self: &mut Program,
```
Comment (kvark) [inline src/front/glsl_new/ast.rs:44]: why is this not just `&mut self`?

Comment (pjoe) [reply src/front/glsl_new/ast.rs:44]: Because I'm still a rust noob :P

Simply copy pasted the external function and made it compile 🤦‍♂️

**Pull request merged**